### PR TITLE
Reject tag values w/ newline character (fixes #3254)

### LINKED ISSUE #3254: TagDistributionMeter blows up if tag values include newline characters
We've observed some users who are passing newline characters in their recorded tag values, which are being returned from the API (for `/issues/releases/<release>/tags/`):

```
name: "build-2609↵"
```

These values are being set as keys for React nodes, which causes them to explode.

### PR DESCRIPTION
**Pull request by benvinegar**
/cc @getsentry/api

### CODE REVIEW

File: src/sentry/models/tagvalue.py [hunk 1]
```diff
@@ -44,6 +49,10 @@ class Meta:
 
     __repr__ = sane_repr('project_id', 'key', 'value')
 
+    @classmethod
+    def is_valid_value(self, value):
+        return TAG_VALUE_RE.match(value)
```
Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: This should return a bool.

Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: Why not just do `return '\n' not in value` ? It'd be more efficient than the regular expression for now.

File: src/sentry/coreapi.py [hunk 2]
```diff
@@ -510,6 +510,15 @@ def validate_data(self, project, data):
                     })
                     continue
 
+                if not TagValue.is_valid_value(v):
+                    self.log.info('Discard invalid tag value: %s', v)
+                    data['errors'].append({
+                        'type': EventError.INVALID_DATA,
+                        'name': 'tags',
+                        'value': pair
```
Comment (mattrobenolt) [inline src/sentry/coreapi.py]: Trailing comma plz.

File: src/sentry/models/tagvalue.py [hunk 3]
```diff
@@ -44,6 +44,10 @@ class Meta:
 
     __repr__ = sane_repr('project_id', 'key', 'value')
 
+    @classmethod
+    def is_valid_value(self, value):
```
Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: `self` is _technically_ wrong in this context, but I see we do this elsewhere erroneously. :) So it's fine. I'll fix all of them in a followup.

**Comment by mattrobenolt** [comment]
:cookie:

File: src/sentry/models/tagvalue.py (hunk 3 above)
Comment (benvinegar) [inline src/sentry/models/tagvalue.py]: "I'll fix it later" is my life motto, 👍

Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: 170f79ad5a992a0653a63c7f50f41d7c62fa69d9 💃

Comment (benvinegar) [inline src/sentry/models/tagvalue.py]: I definitely just borrowed this from `TagKey`

**Pull request merged**